Climbing Upgrade #75

corbinbylsma opened the pull request:
The climbing playerstate now differentiates between webs and ropes, and adjusts physics accordingly. Physics have also been improved.

Commit:
Climbing Upgrade #75

The climbing state now differentiates between webs and ropes, and adjusts physics accordingly. Physics have also been improved.

nickworks commented:
When in a web, it seems like moving right is slower than moving left. I'm not sure where that's coming from. I'm testing in the WeaversWeb level.

I like that moving up or down is now faster, but I think we need to drastically reduce acceleration when the player is climbing. We could switch to linear velocity or maybe clamp the maximum climbing velocity.

Also, lines 112, 116, and 119 have literals that could probably be replaced with constants or variables.

stormwarriors2 committed:
Version 1.2 Hotfix Fix MEs :

First, update this branch from  master

Fix Me : :+1:  Fix the web and rope keep as seperate states instead of a boolean

Fix the onenter to multiply the velocity by an amount, or create a onenterstay for grabbing force and velocity.
OnExit slow the gravity of the player down

Refactor the code (This is optional)

Commit:
Climbing Version 2

Bam
It WORKS now

- The player controller itself now tracks what state the player is currently in. This fixes problems causing jumping in front of climbing volumes to decrease in power for some reason. Can be used for any state, not just climbing
- PlayerStateClmbing features tons of rewriting to achieve essentially the same physics, but is now 100% consistent (up from 70%).
- PlayerStateClimbing is fully documented
- PlayerState has some documentation fixes
- Grab button was changed to E, subject to change.
